src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -262,13 +270,26 @@ public void PreReturn(out bool handledCommand)
 
             public void PostReturn()
             {
+                this.AssertIsForeground();
                 if (this.GetCaretPosition().HasValue)
                 {
                     var closingSnapshotPoint = ClosingPoint.GetPoint(SubjectBuffer.CurrentSnapshot);
 
                     if (closingSnapshotPoint.Position > 0 && HasNoForwardTyping(this.GetCaretPosition().Value, closingSnapshotPoint.Subtract(1)))
                     {
-                        _session.AfterReturn(this, CancellationToken.None);
+                        var context = GetBraceCompletionContext();
+                        if (context == null)
+                        {
+                            return;
+                        }
+
+                        var changesAfterReturn = _service.GetTextChangeAfterReturnAsync(context.Value, CancellationToken.None).WaitAndGetResult(CancellationToken.None);
```

Comment:
i don't love hte inconsistency with how CancellationToken.NOne is treated across all these methods.